config/config.go

```diff
@@ -547,6 +550,30 @@ func (cfg *Manager) IsLoadedFromJSON(t SectionType, name string) bool {
 	return !cfg.undefinedComps[t][name]
 }
 
+func (cfg *Manager) String() (string, error) {
+	var result strings.Builder
+
+	clusterConfig, err := cfg.clusterConfig.ToDisplayJSON()
+	if err != nil {
+		return "", err
+	}
+
+	// ignoring error from WriteString since it will always be nil
+	result.WriteString(fmt.Sprintf("%-20s : %s\n\n", cfg.clusterConfig.ConfigKey(), clusterConfig))
+	for _, stype := range SectionTypes() {
+		for _, s := range cfg.sections[stype] {
+			cfgStr, err := s.ToDisplayJSON()
+			if err != nil {
+				return "", err
+			}
+			result.WriteString(fmt.Sprintf("%-20s : %s\n", s.ConfigKey(), cfgStr))
+		}
+		result.WriteString("\n")
```

Comment:
Rather than printing something by hand, you can build a `jsonConfig` using the DisplayJSONs provided by every component, and print the DefaultJSONMarshal of that. Similar to how `ToJSON`. In fact, the stuff in ToJSON  can probably be re-used by putting most into a parametrized function with an option to calls either component.ToJSON or component.ToDisplayJSON.